### ISSUE TITLE
Write shadow output files to sub directory

### ISSUE THREAD

**Issue by kathapand** [3dgeo-heidelberg/vostok#6]
The shadow output files are currently written to the console's working directory. This can be unfavorable if a large amount of files (depending on the timesteps and period) is written.

I suggest to create a sub-directory, e.g. `shadow_outputs`, [where the shadow output filename is created](https://github.com/3dgeo-heidelberg/vostok/blob/main/src/main.cpp#L136). The alternative would be to have a user-defined path, but I assume it is not necessary to have this flexibility (with the disadvantage of another parameter).

**Comment by bhoefle-3dgeo** [comment]
Added a hard-coded sub-directory for shadow point clouds: https://github.com/3dgeo-heidelberg/vostok/commit/b06c8e01e07e462a33f7c2747dae233cfbc2f17a

**Comment by kathapand** [comment]
The sub-directory is not automatically created if it does not exist. Creation of the directory is added with [d8506cd](https://github.com/3dgeo-heidelberg/vostok/commit/d8506cdfa9c6a990cb01212e5361b2e703ebd814). If the directory exists, the creation will have no effect (see [here](https://en.cppreference.com/w/cpp/filesystem/create_directory)).

The new version is tested and precompiled binaries updated. Closing this issue as solved.

**Comment by bhoefle-3dgeo** [comment]
It is actually done outside the loop where it should belong and only if the parameter for writing output files (value=2) is set. please undo your commit.
See: https://github.com/3dgeo-heidelberg/vostok/blob/d8506cdfa9c6a990cb01212e5361b2e703ebd814/src/main.cpp#L96

**Comment by kathapand** [comment]
Sorry for the poor edit, the commit and compiled version are reverted. I found out the error in my test: my shadow computing parameter was set to `1`. So there is actually an underlying issue I misinterpreted: 

In the previous version files were written anyways, because the `m_computeShadows` condition is not checked when opening the ofstream ([here](https://github.com/3dgeo-heidelberg/vostok/blob/main/src/main.cpp#L151)). It is checked for writing the output ([here]), so that empty files are written if the parameter is not set to `2`.

With the current improvement of creating a subdirectory, empty files are written to `shadow_clouds`, if the directory were to exist already (e.g., due to previous calculations).

I do not know how to elegantly fix it quickly, because the file opening is best done outside the loop, but then the ofstream is not available in the loop, outside the condition's scope. I will leave the issue open for now and can thoroughly check the required edit later.
